### PR TITLE
feat: add attachment support

### PR DESCRIPTION
**Pull request by MarvinJanssen**
## Description

Adds the ability to submit an attachment alongside a transaction. There is currently no indication of this in the transaction confirmation screen. Work in progress. See https://github.com/blockstack/connect/pull/105.

It displays the attachment in a separate row if present. It prints it as ascii if the attachment is composed of only readable characters, otherwise it displays it as a hex string.

The display could use some work. It looks a little messy like any other long parameter (like the zonefile-hash line.)

Screenshot 2021-05-05 at 9 40 16 PM

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No.

## Are documentation updates required?
Yes. TBD.

## Testing information

1. Successful submission of a transaction with attachment.

## Checklist
- [x] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Tag 1 of @hstove or @kyranjamie or @aulneau for review

### CODE REVIEW

**Comment by aulneau** [comment]
This is great, i've tested it locally against an app i'm making and works perfectly!

**Review by aulneau — APPROVED** [review]

**Comment by aulneau** [comment]
@MarvinJanssen feel free to add a changeset (`yarn changeset add`) and I'd be happy to get this merged

**Comment by MarvinJanssen** [comment]
Added a changeset and basic display that is in-line with the others. Let me know. (See the OP.)

File: src/components/transactions/attachment-row.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { useTxState } from '@common/hooks/use-tx-state';
+import { Flex, Box, Text } from '@stacks/ui';
+import React from 'react';
+
+function readable(data: string)
+	{
+	const text = Buffer.from(data,'hex').toString('ascii');
```
Comment (aulneau) [inline src/components/transactions/attachment-row.tsx:7]: is it an ascii string or utf8?

**Comment by aulneau** [comment]
@MarvinJanssen great job! I have three requests for ya:
- run `yarn lint:prettier:fix`
- rebase your commits to be in the format of [convention commits](https://www.conventionalcommits.org/en/v1.0.0/#summary), eg `fix: better regex` instead of `Better regex`
- Can you update the changeset to be a bit more verbose? [here is an example](https://github.com/blockstack/stacks-wallet-web/blob/main/CHANGELOG.md#minor-changes-1), feel free to add the image, too!

Once those are done, I'm happy to merge this as I have QA'ed it, too

File: src/components/transactions/attachment-row.tsx (hunk 1 above)
Comment (MarvinJanssen) [reply src/components/transactions/attachment-row.tsx:7]: I try to force it as ASCII to see it is readable, but we can also try utf8.

**Pull request merged**